turbopack/crates/turbo-tasks-backend/src/kv_backing_storage.rs

```diff
@@ -384,145 +349,187 @@ impl<T: KeyValueDatabase + Send + Sync + 'static> BackingStorage
     }
 }
 
-type OrganizedTaskData = HashMap<
-    TaskId,
-    HashMap<CachedDataItemKey, (Option<CachedDataItemValue>, Option<CachedDataItemValue>)>,
->;
-type ShardedOrganizedTaskData = Vec<OrganizedTaskData>;
+type SerializedTasks = Vec<Vec<(TaskId, Vec<u8>)>>;
 
-fn organize_task_data(updates: Vec<ChunkedVec<CachedDataUpdate>>) -> ShardedOrganizedTaskData {
+fn process_task_data(
```

Comment:
merged organize_task_data + restore_task_data + serialize_task_data and made them all parallelized